Weather-App/Review/App.tsx

```diff
@@ -0,0 +1,139 @@
+
+import './App.css';
+import {useEffect, useState} from 'react';
+import {Input} from "@nextui-org/react";
+import axios from 'axios';
+import { MagnifyingGlassIcon } from '@heroicons/react/24/solid'
+import toast, { Toaster } from 'react-hot-toast';
+import {WeatherData} from './WeatherData'
+
+const notify = (text : string) => toast(text);
+const base_url = "https://weather-app-api-nine.vercel.app/"
+function App() {
+  
+  const [weatherInfo,setInfo] = useState<WeatherData>();
+  const [location, setLocation] = useState("Calgary");
+  const [locationState, setLocationState] = useState("Calgary, Alberta");
+  const [targetURL, setTargetURL] = useState(base_url + "api/WeatherForecast/Weather?lon=-114.065465&lat=51.0460954")
+  const [loading, setLoading] = useState(true);
+
+
+
+  const iconPath = (path : string) => {
+    return `http://openweathermap.org/img/wn/${path}@2x.png`
+  }
+
+  const getLocationURL = () => {
+    return base_url + `api/WeatherForecast/Location?q=${location}`
+  }
+
+  const getWeatherURL = (lat : string, lon : string) => {
+    return base_url + `api/WeatherForecast/Weather?lat=${lat}&lon=${lon}`
+  }
+
+  const getDate = (dateString : number) => {
+    const temp = new Date(dateString * 1000);
+    return temp.toDateString().toString();
+  }
+
+  const getTemperatureRange = (start : number, end : number) => {
+    return `${start}°C to ${end}°C`;
+  }
+
+  useEffect(() => {
+    axios.get(targetURL)
+      .then(res => {
+        //console.log(res)
+        const persons = res.data;
+        setInfo(persons);
+        //console.log(persons)        
+      })
+
+    setLoading(false)
+  },[targetURL])
+
+  const changLocation = (e : React.KeyboardEvent<HTMLInputElement>) => {
+    //console.log(location)
+    if (e.key === 'Enter' && location.length > 0){
+      //
+      const locationURL = getLocationURL();
+      axios.get(locationURL)
+      .then(res => {
+        const locationData = res.data;
+        if (locationData.length > 0){
+          const weatherURL = getWeatherURL(locationData[0].lat, locationData[0].lon)
+          
+          //console.log("weatherURL", weatherURL)
+          if (locationData[0].state !== null && locationData[0].state !== undefined){
+            setLocationState(locationData[0].name + ", " + locationData[0].state)
+          } else {
+            setLocationState(locationData[0].name)
+          }
+          setTargetURL(weatherURL)
+        } else {
+          notify("Location does not exist!")
+        }
+      })
+    }
+
+  }
+
+
+  return (
+    <div id="outside">
+      { loading ? (<h1 className="my-5 text-white">Loading ... </h1>) : (
```

Comment:
it makes it easier to read if you do the loading state outside and return it early

```
if (loading) {
  return <h1 className="my-5 text-white">Loading ... </h1>
}

return (rest of success state code here)

```